code/modules/xenoarcheaology/effects/cannibal.dm

```diff
@@ -0,0 +1,75 @@
+/datum/artifact_effect/cannibalfeeling
+	name = "cannibalfeeling"
+	effect_type = EFFECT_PSIONIC
+	var/list/messages = list("You feel peckish.",
+		"Something doesn't feel right.",
+		"You get a strange feeling in your gut.",
+		"You feel particularly hungry.",
+		"You taste blood.",
+		"There's a strange feeling in the air.",
+		"There's a strange smell in the air.",
+		"The tips of your fingers feel tingly.",
+		"You feel twitchy.",
+		"You feel empty.",
+		"You've got a good feeling about this.",
+		"Your tongue prickles.",
+		"Are they clean?",
+		"You feel weak.",
+		"The ground is getting closer.",
+		"Something is missing.")
+
+	var/list/drastic_messages = list("They look delicious.",
+		"They'll take what's yours!",
+		"They're full of meat.",
+		"What's happening to you?",
+		"Butcher them!",
+		"Feast!")
+
+/datum/artifact_effect/cannibalfeeling/DoEffectTouch(var/mob/user)
+	if(user)
+		if (istype(user, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = user
+			if(H.is_sentient())
+				if(prob(50))
+					if(prob(75))
+						to_chat(H, "<b><font color='red' size='[num2text(rand(1,5))]'>[pick(drastic_messages)]</b></font>")
```

Comment:
Consider alternatively: The artifact applies a _modifier_ that outputs the messages, and stacks duration

Reply:
I used the same system the other 2 message effects do, and modifiers would then need some sort of way to check distance to the artifact, and at that point it's doing the same thing just not localized, and also Very Ugly with anomaly batteries.

Reply:
Well, suppose the modifier lasts only as long as or just slightly longer than the time difference between two process() calls (or w/e it is that's calling this stuff), then the modifier will expire either shortly after the mob leaves the vicinity of the artifact (although possibly spam chat in the case of the first) or if someone is stood around an artifact that's making them think vaddat flesh sounds tasty for like, an hour, the effects are probably gonna persist.